### ISSUE TITLE
Pyright is not understanding TypeVars in `__new__` correctly

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2646]
**Describe the bug**
Pyright appears to be misunderstanding TypeVar usage in `__new__`, producing false errors.

**Expected behavior**
The code to successfully type check, consistent with mypy.

**Screenshots or Code**
The relevant typeshed stub: https://github.com/python/typeshed/blob/master/stdlib/itertools.pyi#L206
```python
# triplewise from more-itertools
def triplewise(iterable: Iterable[T]) -> Iterator[tuple[T, T, T]]:
    "Return overlapping triplets from an iterable"
    # triplewise('ABCDEFG') -> ABC BCD CDE DEF EFG
    for (a, _), (b, c) in itertools.pairwise(itertools.pairwise(iterable)):
        yield a, b, c
```
`itertools.pairwise(iterable)` produces the following error:
```
error: Argument of type "Iterable[T@triplewise]" cannot be assigned to parameter "__iterable" of type "Iterable[_T@__new__]" in function "__new__"
    TypeVar "_T_co@Iterable" is covariant
      Type "T@triplewise" cannot be assigned to type "_T@__new__"
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line (pyright 1.1.91)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

Pyright contained a bug that affected `__new__` methods with a TypeVar that was not scoped to its associated class. This is a rare case, which explains why it hasn't been reported previously.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.192, which I just published. It will also be included in the next release of pylance.
